FreeRTOS/FreeRTOS-Plus-TCP
# Add NetworkDown notification to NetworkInterface.c

Add NetworkDown notification support to STM32FXX network interface.

Description
-----------
NetworkInterface for STM32FXX lacks support for NetworkDown notification.
Current driven only disables MAC interface without notification to higher layers.
Proposed solution add this notification after MAC layer operations.

Test Steps
-----------
1. Run FreeRTOS TCP stack on STM32FXX board with ethernet cable attached.
2. Disconnect ethernet cable.
3. NetworkDown notification will be submitted, higher stack layers will be notified about situation and user callback will be called ( vApplicationIPNetworkEventHook(eNetworkDown) )

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
